// @xelis/sdk/daemon/types.js
export var BalanceType;
(function (BalanceType) {
    BalanceType["Input"] = "input";
    BalanceType["Output"] = "output";
    BalanceType["Both"] = "both";
})(BalanceType || (BalanceType = {}));
export var BlockType;
(function (BlockType) {
    BlockType["Sync"] = "Sync";
    BlockType["Normal"] = "Normal";
    BlockType["Side"] = "Side";
    BlockType["Orphaned"] = "Orphaned";
})(BlockType || (BlockType = {}));
export var RPCMethod;
(function (RPCMethod) {
    RPCMethod["GetVersion"] = "get_version";
    RPCMethod["GetInfo"] = "get_info";
    RPCMethod["GetHeight"] = "get_height";
    RPCMethod["GetTopoHeight"] = "get_topoheight";
    RPCMethod["GetStableHeight"] = "get_stableheight";
    RPCMethod["GetBlockTemplate"] = "get_block_template";
    RPCMethod["GetBlockAtTopoHeight"] = "get_block_at_topoheight";
    RPCMethod["GetBlocksAtHeight"] = "get_blocks_at_height";
    RPCMethod["GetBlockByHash"] = "get_block_by_hash";
    RPCMethod["GetTopBlock"] = "get_top_block";
    RPCMethod["GetNonce"] = "get_nonce";
    RPCMethod["GetNonceAtTopoheight"] = "get_nonce_at_topoheight";
    RPCMethod["HasNonce"] = "has_nonce";
    RPCMethod["GetBalance"] = "get_balance";
    RPCMethod["HasBalance"] = "has_balance";
    RPCMethod["GetBalanceAtTopoHeight"] = "get_balance_at_topoheight";
    RPCMethod["GetAsset"] = "get_asset";
    RPCMethod["GetAssets"] = "get_assets";
    RPCMethod["CountAssets"] = "count_assets";
    RPCMethod["CountTransactions"] = "count_transactions";
    RPCMethod["GetTips"] = "get_tips";
    RPCMethod["P2PStatus"] = "p2p_status";
    RPCMethod["GetDAGOrder"] = "get_dag_order";
    RPCMethod["GetMempool"] = "get_mempool";
    RPCMethod["GetTransaction"] = "get_transaction";
    RPCMethod["GetTransactions"] = "get_transactions";
    RPCMethod["GetBlocksRangeByTopoheight"] = "get_blocks_range_by_topoheight";
    RPCMethod["GetBlocksRangeByHeight"] = "get_blocks_range_by_height";
    RPCMethod["GetAccounts"] = "get_accounts";
    RPCMethod["SubmitBlock"] = "submit_block";
    RPCMethod["SubmitTransaction"] = "submit_transaction";
    RPCMethod["CountAccounts"] = "count_accounts";
    RPCMethod["GetAccountHistory"] = "get_account_history";
    RPCMethod["GetAccountAssets"] = "get_account_assets";
    RPCMethod["GetPeers"] = "get_peers";
    RPCMethod["GetDevFeeThresholds"] = "get_dev_fee_thresholds";
    RPCMethod["GetSizeOnDisk"] = "get_size_on_disk";
    RPCMethod["IsTxExecutedInBlock"] = "is_tx_executed_in_block";
})(RPCMethod || (RPCMethod = {}));
export var RPCEvent;
(function (RPCEvent) {
    RPCEvent["NewBlock"] = "new_block";
    RPCEvent["BlockOrdered"] = "block_ordered";
    RPCEvent["BlockOrphaned"] = "block_orphaned";
    RPCEvent["StableHeightChanged"] = "stable_height_changed";
    RPCEvent["TransactionOrphaned"] = "transaction_orphaned";
    RPCEvent["TransactionAddedInMempool"] = "transaction_added_in_mempool";
    RPCEvent["TransactionExecuted"] = "transaction_executed";
    RPCEvent["TransactionSCResult"] = "transaction_sc_result";
    RPCEvent["NewAsset"] = "new_asset";
    RPCEvent["PeerConnected"] = "peer_connected";
    RPCEvent["PeerDisconnected"] = "peer_disconnected";
    RPCEvent["PeerPeerListUpdated"] = "peer_peer_list_updated";
    RPCEvent["PeerStateUpdated"] = "peer_state_updated";
    RPCEvent["PeerPeerDisconnected"] = "peer_peer_disconnected";
})(RPCEvent || (RPCEvent = {}));
