// @xelis/sdk/daemon/websocket.js
import { RPCEvent, RPCMethod } from './types';
import { WS as BaseWS } from '../lib/websocket';
export class DaemonMethods {
    constructor(ws, prefix = "") {
        this.ws = ws;
        this.prefix = prefix;
    }
    async listenEvent(event, onData) {
        return this.ws.listenEvent(this.prefix + event, onData);
    }
    dataCall(method, params) {
        return this.ws.dataCall(this.prefix + method, params);
    }
    onNewBlock(onData) {
        return this.listenEvent(RPCEvent.NewBlock, onData);
    }
    onTransactionAddedInMempool(onData) {
        return this.listenEvent(RPCEvent.TransactionAddedInMempool, onData);
    }
    onTransactionExecuted(onData) {
        return this.listenEvent(RPCEvent.TransactionExecuted, onData);
    }
    onBlockOrdered(onData) {
        return this.listenEvent(RPCEvent.BlockOrdered, onData);
    }
    onPeerConnected(onData) {
        return this.listenEvent(RPCEvent.PeerConnected, onData);
    }
    onPeerDisconnected(onData) {
        return this.listenEvent(RPCEvent.PeerDisconnected, onData);
    }
    onPeerPeerListUpdated(onData) {
        return this.listenEvent(RPCEvent.PeerPeerListUpdated, onData);
    }
    onPeerPeerDisconnected(onData) {
        return this.listenEvent(RPCEvent.PeerPeerDisconnected, onData);
    }
    onPeerStateUpdated(onData) {
        return this.listenEvent(RPCEvent.PeerStateUpdated, onData);
    }
    onNewAsset(onData) {
        return this.listenEvent(RPCEvent.NewAsset, onData);
    }
    onBlockOrphaned(onData) {
        return this.listenEvent(RPCEvent.BlockOrphaned, onData);
    }
    onTransactionOrphaned(onData) {
        return this.listenEvent(RPCEvent.TransactionOrphaned, onData);
    }
    onStableHeightChanged(onData) {
        return this.listenEvent(RPCEvent.StableHeightChanged, onData);
    }
    getVersion() {
        return this.dataCall(RPCMethod.GetVersion);
    }
    getHeight() {
        return this.dataCall(RPCMethod.GetHeight);
    }
    getTopoHeight() {
        return this.dataCall(RPCMethod.GetTopoHeight);
    }
    getStableHeight() {
        return this.dataCall(RPCMethod.GetStableHeight);
    }
    getBlockTemplate(address) {
        return this.dataCall(RPCMethod.GetBlockTemplate, { address });
    }
    getBlockAtTopoHeight(params) {
        return this.dataCall(RPCMethod.GetBlockAtTopoHeight, params);
    }
    getBlocksAtHeight(params) {
        return this.dataCall(RPCMethod.GetBlocksAtHeight, params);
    }
    getBlockByHash(params) {
        return this.dataCall(RPCMethod.GetBlockByHash, params);
    }
    getTopBlock(params) {
        return this.dataCall(RPCMethod.GetTopBlock, params);
    }
    submitBlock(blockTemplate) {
        return this.dataCall(RPCMethod.SubmitBlock, { block_template: blockTemplate });
    }
    getBalance(params) {
        return this.dataCall(RPCMethod.GetBalance, params);
    }
    hasBalance(params) {
        return this.dataCall(RPCMethod.HasBalance, params);
    }
    getBalanceAtTopoHeight(params) {
        return this.dataCall(RPCMethod.GetBalanceAtTopoHeight, params);
    }
    getInfo() {
        return this.dataCall(RPCMethod.GetInfo);
    }
    getNonce(params) {
        return this.dataCall(RPCMethod.GetNonce, params);
    }
    hasNonce(params) {
        return this.dataCall(RPCMethod.HasNonce, params);
    }
    getAsset(params) {
        return this.dataCall(RPCMethod.GetAsset, params);
    }
    getAssets() {
        return this.dataCall(RPCMethod.GetAssets);
    }
    countAssets() {
        return this.dataCall(RPCMethod.CountAssets);
    }
    countAccounts() {
        return this.dataCall(RPCMethod.CountAccounts);
    }
    countTransactions() {
        return this.dataCall(RPCMethod.CountTransactions);
    }
    submitTransaction(hexData) {
        return this.dataCall(RPCMethod.SubmitTransaction, { data: hexData });
    }
    getTransaction(hash) {
        return this.dataCall(RPCMethod.GetTransaction, { hash });
    }
    p2pStatus() {
        return this.dataCall(RPCMethod.P2PStatus);
    }
    getPeers() {
        return this.dataCall(RPCMethod.GetPeers);
    }
    getMemPool() {
        return this.dataCall(RPCMethod.GetMempool);
    }
    getTips() {
        return this.dataCall(RPCMethod.GetTips);
    }
    getDAGOrder(params) {
        return this.dataCall(RPCMethod.GetDAGOrder, params);
    }
    getBlocksRangeByTopoheight(params) {
        return this.dataCall(RPCMethod.GetBlocksRangeByTopoheight, params);
    }
    getBlocksRangeByHeight(params) {
        return this.dataCall(RPCMethod.GetBlocksRangeByHeight, params);
    }
    getTransactions(txHashes) {
        return this.dataCall(RPCMethod.GetTransactions, { tx_hashes: txHashes });
    }
    getAccountHistory(params) {
        return this.dataCall(RPCMethod.GetAccountHistory, params);
    }
    getAccountAssets(address) {
        return this.dataCall(RPCMethod.GetAccountAssets, { address });
    }
    getAccounts(params) {
        return this.dataCall(RPCMethod.GetAccounts, params);
    }
    isTxExecutedInBlock(params) {
        return this.dataCall(RPCMethod.IsTxExecutedInBlock, params);
    }
    getDevFeeThresholds() {
        return this.dataCall(RPCMethod.GetDevFeeThresholds);
    }
    getSizeOnDisk() {
        return this.dataCall(RPCMethod.GetSizeOnDisk);
    }
}
export class WS extends BaseWS {
    constructor() {
        super();
        this.methods = new DaemonMethods(this);
    }
}
export default WS;
