// @xelis/sdk/wallet/types.js
export var RPCMethod;
(function (RPCMethod) {
    RPCMethod["GetVersion"] = "get_version";
    RPCMethod["GetNetwork"] = "get_network";
    RPCMethod["GetNonce"] = "get_nonce";
    RPCMethod["GetTopoheight"] = "get_topoheight";
    RPCMethod["GetAddress"] = "get_address";
    RPCMethod["SplitAddress"] = "split_address";
    RPCMethod["Rescan"] = "rescan";
    RPCMethod["GetBalance"] = "get_balance";
    RPCMethod["HasBalance"] = "has_balance";
    RPCMethod["GetTrackedAssets"] = "get_tracked_assets";
    RPCMethod["GetAssetPrecision"] = "get_asset_precision";
    RPCMethod["GetTransaction"] = "get_transaction";
    RPCMethod["BuildTransaction"] = "build_transaction";
    RPCMethod["ListTransactions"] = "list_transactions";
    RPCMethod["IsOnline"] = "is_online";
    RPCMethod["SignData"] = "sign_data";
    RPCMethod["EstimateFees"] = "estimate_fees";
})(RPCMethod || (RPCMethod = {}));
export var Permission;
(function (Permission) {
    Permission[Permission["Ask"] = 0] = "Ask";
    Permission[Permission["AcceptAlways"] = 1] = "AcceptAlways";
    Permission[Permission["DenyAlways"] = 2] = "DenyAlways";
})(Permission || (Permission = {}));
